src/defrag.c

```diff
@@ -963,9 +970,15 @@ void activeDefragCycle(void) {
 
     /* Once a second, check if the fragmentation justfies starting a scan
      * or making it more aggressive. */
-    run_with_period(1000) {
+    if (!server.active_defrag_configuration_changed) {
+        run_with_period(1000) {
+            computeDefragCycles();
+        }
+    } else {
         computeDefragCycles();
+        server.active_defrag_configuration_changed = 0;
```

Comment:
Does this mean that once the threshold is changed we go to defrag immediately, instead of waiting up to 1 second?

Reply:
yes, i am hesitating whether to add this. On the one hand, i think diff is a bit ugly. On the other hand, i think real-time is better.

Reply:
I prefer to leave it as is, just as changing `save` config doesn't immediately save.

Reply:
it is actually just advances the compute, which is still different from save i think.

Reply:
i don't mind waiting a second, and i don't think i mind making it immediate. but the code is indeed ugly.
how about just calling computeDefragCycles again (without and if-else).
i.e. in the rare chance that the time will hit us at exactly the same moment as the config, we'll run it twice.

Reply:
right, that is a good idea, the code now looks much better.